lib/runCall.js

```diff
@@ -160,6 +160,12 @@ module.exports = function (opts, cb) {
         var totalGas = results.gasUsed
         if (!results.runState.vmError) {
           var returnFee = results.return.length * fees.createDataGas.v
+
+          // avoid BN assertion failure when returnFee is greater than 0x4000000
+          if (returnFee > gasLimit.toNumber()) {
+            returnFee = gasLimit.toNumber() + 1
```

Comment:
Should this instead emit an "internal error" here? It would be more straightfoward as this one is causing an OOG below, but it is not clear without reading the next `if`.